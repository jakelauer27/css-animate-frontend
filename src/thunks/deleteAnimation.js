import { deleteMyAnimation } from '../actions/actions'

export const deleteAnimation = (user_id, animation_id) => {
  return async (dispatch) => {
    try {
      const response = await fetch(`https://css-animate-backend.herokuapp.com/api/users/${user_id}/animations/${animation_id}`, {
        method: 'DELETE',
      })
      const data = await response.json()
      dispatch(deleteMyAnimation(animation_id))
      return data
    } catch (error) {
      return new Error(error)
    }
  } 
}