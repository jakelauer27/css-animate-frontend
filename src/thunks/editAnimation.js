import { editMyAnimation } from '../actions/actions'

export const editAnimation = (user_id, animation_id, animation) => {
  return async (dispatch) => {
    try {
      const response = await fetch(`http://localhost:3000/api/users/${user_id}/animations/${animation_id}`, {
        method: 'PUT',
        body: JSON.stringify({user_id, animation_id, ani_name: animation.name, properties: animation.properties, keyframes: animation.keyframes}),
        headers: {
          'Content-Type': 'application/json'
        }
      })
      const data = response.json()
      return data
    } catch(error) {
      return new Error(error)
    }
  }
}