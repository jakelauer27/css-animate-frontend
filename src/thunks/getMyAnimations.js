import { loadMyAnimations } from '../actions/actions'

export const getMyAnimations = (user_id) => {
  return async (dispatch) => {
    try {
      const response = await fetch(`http://localhost:3000/api/users/${user_id}/animations`)
      const data = await response.json()
      dispatch(loadMyAnimations(data.data))
    } catch(error) {
      return new Error(error)
    }
  }
}