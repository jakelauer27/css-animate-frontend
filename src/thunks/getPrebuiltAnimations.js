import { loadPrebuiltAnimations } from '../actions/actions'

export const getPrebuiltAnimations = () => {
  return async (dispatch) => {
    try {
      const response = await fetch("https://css-animate-backend.herokuapp.com/api/users/animations")
      const data = await response.json()
      dispatch(loadPrebuiltAnimations(data.data))
    } catch(error) {
      return new Error(error)
    }
  }
}