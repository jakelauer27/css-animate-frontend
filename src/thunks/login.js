import { signIn } from '../actions/actions'

export const loginUser = (user) => {
  return async (dispatch) => {
    try {
      const response = await fetch("https://css-animate-backend.herokuapp.com/api/users/", {
        method: 'POST',
        body: JSON.stringify(user),
        headers: {
          'Content-Type': 'application/json'
        }
      })
      const data = await response.json()
      dispatch(signIn(data.data))
      return data
    } catch(error) {
      return new Error(error)
    }
  }
}