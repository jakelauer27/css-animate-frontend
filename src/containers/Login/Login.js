import React, { Component } from 'react'
import { Redirect, Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { PropTypes } from 'prop-types'
import { loginUser } from '../../thunks/login'
import { getMyAnimations } from '../../thunks/getMyAnimations'

export class Login extends Component {
  constructor() {
    super()
    this.state = {
      email: '',
      password: '',
      incorrect: false
    }
  }

  handleChange = (e) => {
    const { value, name } = e.target
    this.setState({
      [name]: value
    })
  }

  async handleSubmit(e) {
    e.preventDefault()
    const { getMyAnimations, loginUser } = this.props
    const { email, password } = this.state
    const success = await loginUser({email, password})
    if(success.status === 'success') {
      await getMyAnimations(success.data.id)
    } else {
      this.setState({incorrect: true})
    }
  }

  render() {
    const { email, password } = this.state
    const { user } = this.props
    if( user.name ) {
      return (
        <Redirect to='/properties' />
      )
    } 
    
    return (
      <div className='login-container'>
        <div className='infoPopup-overlay'></div>
        <div className='form-container'>
          <h2 className='login-title'>Login</h2>
          <form className='login-form' onSubmit={(e) => this.handleSubmit(e)}>
            <input 
              className='form-input email'
              type='text' 
              placeholder='email' 
              value={email} 
              name='email'
              onChange={this.handleChange}>
            </input>
            <input 
              className='form-input password'
              type='password' 
              placeholder='password' 
              value={password}
              name='password'
              onChange={this.handleChange}>
            </input>
            <button>Login</button>
          </form>
          <div className='new-account-container'>
            <h3 className={`incorrect-login ${this.state.incorrect && 'incorrect'}`}>username or password is incorrect</h3>
            <h3>Don't have an account?</h3>
            <Link className='form-link' to='/properties/signup' >Create New Account</Link>
            <Link className='form-link continue-as-guest' to='/properties' >Continue as Guest</Link>
          </div>
        </div>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({ 
  loginUser: (user) => dispatch(loginUser(user)),
  getMyAnimations: (id) => dispatch(getMyAnimations(id))
})

export const mapStateToProps = (state) => ({ 
  user: state.user
})

Login.propTypes = {
  loginUser: PropTypes.func.isRequired,
  user: PropTypes.object,
  getMyAnimations: PropTypes.func.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(Login)