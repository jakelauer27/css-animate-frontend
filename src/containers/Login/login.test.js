import { Login, mapDispatchToProps, mapStateToProps } from './Login'
import React from 'react'
import { shallow } from 'enzyme'
import { loginUser } from '../../thunks/login'
import { getMyAnimations } from '../../thunks/getMyAnimations'

jest.mock('../../thunks/login')
jest.mock('../../thunks/getMyAnimations')

describe('Login Component', () => {
  let loginUserMock
  let mockGetMyAnimations
  let wrapper
  
  beforeEach(() => {
    mockGetMyAnimations = jest.fn()
    loginUserMock = jest.fn().mockImplementation((param) => {
      return {status: 'success', data: {id: 1}}
    })
    wrapper = shallow(<Login loginUser={loginUserMock} user={{name: null}} getMyAnimations={mockGetMyAnimations}/>)
  })
  
  it('matches the snapshot ', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
  it('should match the snapshot when user is logged in', () => {
    wrapper = shallow(<Login loginUser={loginUserMock} user={{name: 'jake'}} getMyAnimations={mockGetMyAnimations}/>)

    expect(wrapper).toMatchSnapshot()
  })
  
  it('calls handleChange when e-mail is changed', () => {
    const spy = spyOn(wrapper.instance(), 'handleChange')
    wrapper.instance().forceUpdate()
    const mockEvent = { target: { value: 'something@gmail.com' } }
    wrapper.find('.email').simulate('change', mockEvent)
    expect(spy).toHaveBeenCalled()
  })
  
  it('calls handleChange when password is changed', () => {
    const spy = spyOn(wrapper.instance(), 'handleChange')
    wrapper.instance().forceUpdate()
    const mockEvent = { target: { value: 'password' } }
    wrapper.find('.password').simulate('change', mockEvent)
    expect(spy).toHaveBeenCalled()
    
  })
  
  it('updates state when handleChange is called', () => {
    const mockEvent = { target: { name: 'email', value: 'something@gmail.com' } }
    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state('email')).toBe('something@gmail.com')
  })
  
  it('calls handleSubmit onSubmit of the form', () => { 
    const spy = spyOn(wrapper.instance(), 'handleSubmit')
    const mockEvent = { preventDefault: jest.fn() }
    wrapper.instance().forceUpdate()
    wrapper.find('form').simulate('submit', mockEvent)
    expect(spy).toHaveBeenCalled()
  })
  
  describe('handleSubmit', () => {
    const mockEvent = {preventDefault: jest.fn()}

    beforeEach(() => {
      wrapper.state().email = 'jake@gmail.com'
      wrapper.state().password = 'jake'
    })

    it('Should call login user from props', async () => {
      const expected = {"email": "jake@gmail.com", "password": "jake"}

      await wrapper.instance().handleSubmit(mockEvent)
      expect(loginUserMock).toHaveBeenCalledWith(expected)      
    })

    it('should call getMyAnimations with the correct params', async () => {
      await wrapper.instance().handleSubmit(mockEvent)

      expect(mockGetMyAnimations).toHaveBeenCalledWith(1)
    })
    
    it('should set state incorrect to true if incorrect login', async () => {
      loginUserMock = jest.fn().mockImplementation((param) => {
        return {status: 'failure', data: {id: 1}}
      })
      wrapper = shallow(<Login loginUser={loginUserMock} user={{name: null}} getMyAnimations={mockGetMyAnimations}/>)
      await wrapper.instance().handleSubmit(mockEvent)
      expect(wrapper.state().incorrect).toEqual(true)
    })
  })

  describe('handleChange', () => {
    
    it('should update state', () => {
      const mockEvent = {target: {value: 'jake', name: 'password'}}
      const expected = 'jake'
      wrapper.instance().handleChange(mockEvent)

      expect(wrapper.state().password).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {

    it('calls dispatch with a signIn action when loginUser is called', async () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = await loginUser({name: 'Justin', email: 'J@aol.com', id: 1, password: 'password'})
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.loginUser({name: 'Justin', email: 'J@aol.com', id: 1, password: 'password'})
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('calls dispatch with a getMyAnimations action when getMyAnimations is called', async () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = await getMyAnimations(1)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.getMyAnimations(1)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })

  describe('mapStateToProps', () => {
    
    it('should return a props object with a user property', () => {
      const mockState = {user: {name: 'jake', id: 1}}
      const expected = {user: {name: 'jake', id: 1}}

      expect(mapStateToProps(mockState)).toEqual(expected)
    })
  })
  
}) 
