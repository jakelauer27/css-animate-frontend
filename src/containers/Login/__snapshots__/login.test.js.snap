// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component matches the snapshot  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    getMyAnimations={[MockFunction]}
    loginUser={[MockFunction]}
    user={
      Object {
        "name": null,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="infoPopup-overlay"
        />,
        <div
          className="form-container"
        >
          <h2
            className="login-title"
          >
            Login
          </h2>
          <form
            className="login-form"
            onSubmit={[Function]}
          >
            <input
              className="form-input email"
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="text"
              value=""
            />
            <input
              className="form-input password"
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            />
            <button>
              Login
            </button>
          </form>
          <div
            className="new-account-container"
          >
            <h3
              className="incorrect-login false"
            >
              username or password is incorrect
            </h3>
            <h3>
              Don't have an account?
            </h3>
            <Link
              className="form-link"
              replace={false}
              to="/properties/signup"
            >
              Create New Account
            </Link>
            <Link
              className="form-link continue-as-guest"
              replace={false}
              to="/properties"
            >
              Continue as Guest
            </Link>
          </div>
        </div>,
      ],
      "className": "login-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "infoPopup-overlay",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="login-title"
            >
              Login
            </h2>,
            <form
              className="login-form"
              onSubmit={[Function]}
            >
              <input
                className="form-input email"
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="text"
                value=""
              />
              <input
                className="form-input password"
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
              />
              <button>
                Login
              </button>
            </form>,
            <div
              className="new-account-container"
            >
              <h3
                className="incorrect-login false"
              >
                username or password is incorrect
              </h3>
              <h3>
                Don't have an account?
              </h3>
              <Link
                className="form-link"
                replace={false}
                to="/properties/signup"
              >
                Create New Account
              </Link>
              <Link
                className="form-link continue-as-guest"
                replace={false}
                to="/properties"
              >
                Continue as Guest
              </Link>
            </div>,
          ],
          "className": "form-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
              "className": "login-title",
            },
            "ref": null,
            "rendered": "Login",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="form-input email"
                  name="email"
                  onChange={[Function]}
                  placeholder="email"
                  type="text"
                  value=""
                />,
                <input
                  className="form-input password"
                  name="password"
                  onChange={[Function]}
                  placeholder="password"
                  type="password"
                  value=""
                />,
                <button>
                  Login
                </button>,
              ],
              "className": "login-form",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form-input email",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "email",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form-input password",
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login",
                },
                "ref": null,
                "rendered": "Login",
                "type": "button",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="incorrect-login false"
                >
                  username or password is incorrect
                </h3>,
                <h3>
                  Don't have an account?
                </h3>,
                <Link
                  className="form-link"
                  replace={false}
                  to="/properties/signup"
                >
                  Create New Account
                </Link>,
                <Link
                  className="form-link continue-as-guest"
                  replace={false}
                  to="/properties"
                >
                  Continue as Guest
                </Link>,
              ],
              "className": "new-account-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "username or password is incorrect",
                  "className": "incorrect-login false",
                },
                "ref": null,
                "rendered": "username or password is incorrect",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Don't have an account?",
                },
                "ref": null,
                "rendered": "Don't have an account?",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Create New Account",
                  "className": "form-link",
                  "replace": false,
                  "to": "/properties/signup",
                },
                "ref": null,
                "rendered": "Create New Account",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Continue as Guest",
                  "className": "form-link continue-as-guest",
                  "replace": false,
                  "to": "/properties",
                },
                "ref": null,
                "rendered": "Continue as Guest",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="infoPopup-overlay"
          />,
          <div
            className="form-container"
          >
            <h2
              className="login-title"
            >
              Login
            </h2>
            <form
              className="login-form"
              onSubmit={[Function]}
            >
              <input
                className="form-input email"
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="text"
                value=""
              />
              <input
                className="form-input password"
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
              />
              <button>
                Login
              </button>
            </form>
            <div
              className="new-account-container"
            >
              <h3
                className="incorrect-login false"
              >
                username or password is incorrect
              </h3>
              <h3>
                Don't have an account?
              </h3>
              <Link
                className="form-link"
                replace={false}
                to="/properties/signup"
              >
                Create New Account
              </Link>
              <Link
                className="form-link continue-as-guest"
                replace={false}
                to="/properties"
              >
                Continue as Guest
              </Link>
            </div>
          </div>,
        ],
        "className": "login-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "infoPopup-overlay",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="login-title"
              >
                Login
              </h2>,
              <form
                className="login-form"
                onSubmit={[Function]}
              >
                <input
                  className="form-input email"
                  name="email"
                  onChange={[Function]}
                  placeholder="email"
                  type="text"
                  value=""
                />
                <input
                  className="form-input password"
                  name="password"
                  onChange={[Function]}
                  placeholder="password"
                  type="password"
                  value=""
                />
                <button>
                  Login
                </button>
              </form>,
              <div
                className="new-account-container"
              >
                <h3
                  className="incorrect-login false"
                >
                  username or password is incorrect
                </h3>
                <h3>
                  Don't have an account?
                </h3>
                <Link
                  className="form-link"
                  replace={false}
                  to="/properties/signup"
                >
                  Create New Account
                </Link>
                <Link
                  className="form-link continue-as-guest"
                  replace={false}
                  to="/properties"
                >
                  Continue as Guest
                </Link>
              </div>,
            ],
            "className": "form-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
                "className": "login-title",
              },
              "ref": null,
              "rendered": "Login",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="form-input email"
                    name="email"
                    onChange={[Function]}
                    placeholder="email"
                    type="text"
                    value=""
                  />,
                  <input
                    className="form-input password"
                    name="password"
                    onChange={[Function]}
                    placeholder="password"
                    type="password"
                    value=""
                  />,
                  <button>
                    Login
                  </button>,
                ],
                "className": "login-form",
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-input email",
                    "name": "email",
                    "onChange": [Function],
                    "placeholder": "email",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-input password",
                    "name": "password",
                    "onChange": [Function],
                    "placeholder": "password",
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Login",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": "button",
                },
              ],
              "type": "form",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3
                    className="incorrect-login false"
                  >
                    username or password is incorrect
                  </h3>,
                  <h3>
                    Don't have an account?
                  </h3>,
                  <Link
                    className="form-link"
                    replace={false}
                    to="/properties/signup"
                  >
                    Create New Account
                  </Link>,
                  <Link
                    className="form-link continue-as-guest"
                    replace={false}
                    to="/properties"
                  >
                    Continue as Guest
                  </Link>,
                ],
                "className": "new-account-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "username or password is incorrect",
                    "className": "incorrect-login false",
                  },
                  "ref": null,
                  "rendered": "username or password is incorrect",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Don't have an account?",
                  },
                  "ref": null,
                  "rendered": "Don't have an account?",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Create New Account",
                    "className": "form-link",
                    "replace": false,
                    "to": "/properties/signup",
                  },
                  "ref": null,
                  "rendered": "Create New Account",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Continue as Guest",
                    "className": "form-link continue-as-guest",
                    "replace": false,
                    "to": "/properties",
                  },
                  "ref": null,
                  "rendered": "Continue as Guest",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Login Component should match the snapshot when user is logged in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    getMyAnimations={[MockFunction]}
    loginUser={[MockFunction]}
    user={
      Object {
        "name": "jake",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "push": false,
      "to": "/properties",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "push": false,
        "to": "/properties",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
